contribution_evaluation.py

```
@@ -0,0 +1,221 @@
+        if not filename.endswith(('.cpp', '.h')):
+            continue
+        # get the content of file by its url
+        file_content = get_file_content(change['contents_url'])
+        if not file_content:
+            continue
+        # extract changed functions from patch
+        added_functions, modified_functions = extract_changed_functions(patch, file_content,filename)
+
+        # set the workload
+        for function in added_functions + modified_functions:
+            function_name = function["function_name"]
```

Ruff (0.8.2)

Local variable `function_name` is assigned to but never used

Remove assignment to unused variable `function_name`

(F841)